solc/CommandLineInterface.cpp

```diff
@@ -1776,14 +1776,20 @@ bool CommandLineInterface::link()
 				return false;
 			}
 
-			string name(it, it + placeholderSize);
-			if (librariesReplacements.count(name))
+			if (*(it + 1) != '_' || *(it + placeholderSize - 2) != '_' || *(it + placeholderSize - 1) != '_')
 			{
-				string hexStr(toHex(librariesReplacements.at(name).asBytes()));
+				++it;
+				continue;
```

Comment:
So this just ignores the error?

Reply:
Yeah. It assumes that it was not a linker reference and continues.

I can make it report an error instead if you think it's a better idea. I originally wanted to make it more robust against weird input but just rejecting the file and telling the user that it's not a binary file might be a better idea.

Reply:
I have updated the PR to report an error instead. Now it will report this:

```
Error in binary object file /tmp/link2.bin at position 18
"_2.sol:C =======
Binary:
608060405234801" is not a valid link reference.
```

when you try to link a file that looks like this:
```
======= /tmp/test_2.sol:C =======
Binary:
608060405234801561001057600080fd5b506101a4806100206000396000f3fe608060405234801561001057600080fd5b506004361061002b5760003560e01c8063c298557814610030575b600080fd5b61003861004e565b604051610045919061011f565b60405180910390f35b6100566100e1565b600073__$6fad237b9cb2f039953c1fb84c4693db8d$__905073__$6fad237b9cb2f039953c1fb84c4693db8d$__63c29855786040518163ffffffff1660e01b815260040160006040518083038186803b1580156100b357600080fd5b505af41580156100c7573d6000803e3d6000fd5b505050506040518060200160405280600181525091505090565b6040518060200160405280600081525090565b60208201600082015161010a6000850182610110565b50505050565b6101198161013a565b82525050565b600060208201905061013460008301846100f4565b92915050565b600081905091905056fea264697066735822122018af78cc6f7227a7edeec8b082453ed805d747299d95845e905bec42ac7bc23864736f6c63782c302e372e362d646576656c6f702e323032302e31312e31392b636f6d6d69742e36663261313332332e6d6f64005d

// $6fad237b9cb2f039953c1fb84c4693db8d$ -> /tmp/test_2.sol:L
// $6fad237b9cb2f039953c1fb84c4693db8d$ -> /tmp/test_2.sol:L

======= /tmp/test_2.sol:L =======
Binary:
60ae610024600b82828239805160001a607314601757fe5b30600052607381538281f3fe730000000000000000000000000000000000000000301460806040526004361060335760003560e01c8063c2985578146038575b600080fd5b818015604357600080fd5b50604a604c565b005b56fea264697066735822122020feea40d2730afe6d877470518bf9c106eadef60b57afbe973c705b90e0ad3c64736f6c63782c302e372e362d646576656c6f702e323032302e31312e31392b636f6d6d69742e36663261313332332e6d6f64005d
```

Reply:
I have also changed the reported position to the first character of the placeholder (previously it would report the last position in file if it ended before the placeholder did),